ooniprobe/Utility/UIDevice-DeviceInfo.m

```diff
@@ -0,0 +1,25 @@
+#include <sys/sysctl.h>
+#import "UIDevice-DeviceInfo.h"
+
+@implementation UIDevice (DeviceInfo)
+
+- (NSString *)getSysInfoByName:(char *)typeSpecifier
+{
+    size_t size;
+    sysctlbyname(typeSpecifier, NULL, &size, NULL, 0);
```

Comment:
`sysctlbyname` returns `0` on success and nonzero on failure. Please check the return value and `abort` if it returns a failure. I assume this kind of functions should not fail until we get the params right.

Reply:
Please, also `abort` if the returned value is `0` and `size` is `0`

Reply:
added other exception, I don't know how to call it

Reply:
`parse_error`

Reply:
Ah, I see what you mean. You're using ObjectiveC exceptions.

To call `abort` just do `abort();`

Reply:
So should I use abort or Objc exceptions? what is suggested?

Reply:
I'm a bit out of my depth here. My understanding is that `abort` should not call `malloc`. I do not know what happens when raising an ObjC exception. My sense is that, if `malloc` fails by returning `NULL`, then the only sane thing is to crash. So, even without knowing about the ObjC exception mechanism details, I'd say it's fine to call `abort` here. (I also checked whether `_exit` would be a better alternative, but the Darwin manual page mentions too many things that `_exit` does, compared to `abort`, so I'm leaning towards `abort` here.)

Reply:
can we resolve this?

Reply:
Yes, let's use `abort`.